modules/geo-layers/src/tile-3d-layer/tile-3d-layer.js

```diff
@@ -127,11 +135,13 @@ export default class Tile3DLayer extends CompositeLayer {
   }
 
   _updateTileset(tileset3d) {
-    const {timeline, viewport} = this.context;
-    if (!timeline || !viewport || !tileset3d) {
+    const {timeline} = this.context;
+    const {activatedViewports = {}} = this.state;
+    const viewportsNumber = Object.keys(activatedViewports).length;
+    if (!timeline || !viewportsNumber || !tileset3d) {
       return;
     }
-    const frameNumber = tileset3d.update(viewport);
+    const frameNumber = tileset3d.update(Object.values(activatedViewports));
```

Comment:
You need to clear `activatedViewports` here after use, otherwise removed viewports will still persist.

Reply:
Tests fails if I do `this.setState({activatedViewports: {}});` here
TAP version 13
# deck.gl
# Tile3DLayer
# inital load
not ok 1 Renders sub layers
  ---
    operator: ok
    expected: true
    actual:   undefined
    stack: |-
      Error: Renders sub layers
          at Test.assert [as _assert] (http://localhost:5000/test-browser.js:269191:54)
          at Test.bound [as _assert] (http://localhost:5000/test-browser.js:269043:32)
          at Test.assert (http://localhost:5000/test-browser.js:269310:10)
          at Test.bound [as ok] (http://localhost:5000/test-browser.js:269043:32)
          at Object.onAfterUpdate (http://localhost:5000/test-browser.js:299855:13)
          at runLayerTestPostUpdateCheck (http://localhost:5000/test-browser.js:44435:14)
          at testLayerAsync (http://localhost:5000/test-browser.js:44355:7)
          at async Object.onLoad (http://localhost:5000/test-browser.js:299889:7)
  ...

Reply:
`setState` triggers a new layer update. You can use `this.state.activeViewports = {}`

Reply:
Done

Reply:
But tests don't pass. I removed this line.

Reply:
If tests don't pass then you need to figure out why. Removing the line doesn't make it correct.

Reply:
Done